src/werkzeug/urls.py

```diff
@@ -541,7 +566,7 @@ def url_quote_plus(string, charset="utf-8", errors="strict", safe=""):
     return url_quote(string, charset, errors, safe + " ", "+").replace(" ", "+")
 
 
-def url_unparse(components):
+def url_unparse(components: Tuple[str, str, str, str, str]) -> str:
```

Comment:
I'm confused as to the return type of this function – in the docstring here it states that the return type is a string (`str`) whereas the relevant docstring states that the return type of `BaseURL.to_url` (which returns the output of this function) is either bytes (`bytes`) or string (`str`).

Reply:
I think it can either return `bytes` or `str` depending on whether the components are all bytes or all str. I think,
```python
def url_unparse(components: Tuple[AnyStr, AnyStr, AnyStr, AnyStr, AnyStr]) -> AnyStr:
```
is the right typing...

Reply:
See URLPartsTuple usage in #1910 (not sure about the name).